### PR TITLE
Remove reference type Graph

### PR DESCRIPTION
**Pull request by fortmarek**
### Short description 📝

In this PR I have:
- removed `Graph` and all its related classes (mostly nodes)
- migrated mainly tests that were still using `Graph` instead of `ValueGraph`

Once this is merged, I think we can also drop the `Value` prefix as it's not really necessary since we now have only one `Graph` model?

Anyways, this also means there are basically no more warnings left and that will make a difference 🙌 

Having value graph everywhere will also make a lot of new features easier to implement, I am really glad that this has been done - thanks @kwridan and @pepibumur for doing this (I just took what was left to be done and brought it over the finish line 🏁)

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/TuistCore/ValueGraph/ValueGraphTraverser.swift [hunk 1]
```diff
@@ -262,7 +262,13 @@ public class ValueGraphTraverser: GraphTraversing {
 
         // AppClip dependencies
         if target.target.isAppClip {
-            let path = try SDKNode.appClip(status: .required).path
+            let path = try SystemFrameworkMetadataProvider().loadMetadata(
```
Comment (pepibumur) [inline Sources/TuistCore/ValueGraph/ValueGraphTraverser.swift:265]: Shall we make the metadata provider a property in this class?

**Review by pepibumur — APPROVED** [review]
I can't stress enough how important this is for the project. Thank you very much for doing it. If CI is happy with the changes we should be able to merge confidently

File: Sources/TuistCore/ValueGraph/ValueGraphTraverser.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistCore/ValueGraph/ValueGraphTraverser.swift:265]: makes sense, let me change that 👍

**Comment by kwridan** [comment]
Thanks for doing this @fortmarek! sorry I didn't get a chance to review this earlier. I spotted a possible regression with how the graph is loaded for the graph and lint commands. The `graph` command will sadly fail in case it is run within a project that has a `.project(target:path:)` dependency.

`ModelLoader` loads individual models and not the entire set of models, the paradigm is slightly different when using `ValueGraphLoader` that expects all the models have been loaded ahead of making the graph _(this was done to support all the new features we've been using in `Generator` such as concurrent loading, mappers, etc...)_.

Will take a look at making a fix for this.

**Pull request merged**